quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/CachingPropertyMappers.java

```diff
@@ -21,61 +24,71 @@ private CachingPropertyMappers() {
-                        .build(),
-                fromOption(CachingOptions.CACHE_CONFIG_FILE)
-                        .mapFrom("cache")
-                        .to("kc.spi-connections-infinispan-quarkus-config-file")
-                        .transformer(CachingPropertyMappers::resolveConfigFile)
-                        .paramLabel("file")
-                        .build(),
-                fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_ENABLED)
-                        .build(),
-                fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_KEYSTORE.withRuntimeSpecificDefault(getDefaultKeystorePathValue()))
-                        .paramLabel("file")
-                        .build(),
-                fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_KEYSTORE_PASSWORD)
-                        .paramLabel("password")
-                        .isMasked(true)
-                        .build(),
-                fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_TRUSTSTORE.withRuntimeSpecificDefault(getDefaultTruststorePathValue()))
-                        .paramLabel("file")
-                        .build(),
-                fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_TRUSTSTORE_PASSWORD)
-                        .paramLabel("password")
-                        .isMasked(true)
-                        .build(),
-                fromOption(CachingOptions.CACHE_REMOTE_HOST)
-                        .paramLabel("hostname")
-                        .build(),
-                fromOption(CachingOptions.CACHE_REMOTE_PORT)
-                        .isEnabled(CachingPropertyMappers::remoteHostSet, CachingPropertyMappers.REMOTE_HOST_SET)
-                        .paramLabel("port")
-                        .build(),
-                fromOption(CachingOptions.CACHE_REMOTE_TLS_ENABLED)
-                        .isEnabled(CachingPropertyMappers::remoteHostSet, CachingPropertyMappers.REMOTE_HOST_SET)
-                        .build(),
-                fromOption(CachingOptions.CACHE_REMOTE_USERNAME)
-                        .isEnabled(CachingPropertyMappers::remoteHostSet, CachingPropertyMappers.REMOTE_HOST_SET)
-                        .paramLabel("username")
-                        .build(),
-                fromOption(CachingOptions.CACHE_REMOTE_PASSWORD)
-                        .isEnabled(CachingPropertyMappers::remoteHostSet, CachingPropertyMappers.REMOTE_HOST_SET)
-                        .paramLabel("password")
-                        .isMasked(true)
-                        .build(),
-
-                fromOption(CachingOptions.CACHE_METRICS_HISTOGRAMS_ENABLED)
-                        .isEnabled(MetricsPropertyMappers::metricsEnabled, MetricsPropertyMappers.METRICS_ENABLED_MSG)
-                        .build(),
-
+        PropertyMapper<?>[] maxCountMappers = Stream.of(CachingOptions.MAX_COUNT_OPTIONS)
+              .map(opt ->
+                    fromOption(opt)
+                          .isEnabled(InfinispanUtils::isEmbeddedInfinispan)
```

Comment:
local caches option should always be enabled. 
Probably it is cleaner to add 2 new arrays in `InfinispanConnectionProvider`
* local bounded caches -> always available to be configured
* clustered bounded caches -> only when in "embedded mode"

Reply:
Currently the `keycloak-config-api` module doesn't have a dependency on `keycloak-model-infinispan` and has minimal dependencies in general, so I'm not sure if we can add this?

Reply:
oh :( having the arrays in CacheOption should be fine.